### PR TITLE
Avoid panic on Kibana telemetry API call

### LINKED ISSUE #3709: TestTelemetry is flaky
A runtime exception happened here: https://github.com/elastic/cloud-on-k8s/blob/ff40a5c3c3f92b1e9a10b5637b275a559413983a/test/e2e/kb/telemetry_test.go#L60, see logs from https://devops-ci.elastic.co/job/cloud-on-k8s-e2e-tests-ocp/225/consoleFull :

```
07:08:55  {"Time":"2020-09-04T05:08:54.819206327Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/test/e2e/kb","Test":"TestTelemetry/Kibana_should_be_able_to_connect_to_Elasticsearch","Output":"=== RUN   TestTelemetry/Kibana_should_be_able_to_connect_to_Elasticsearch\n"}
07:08:55  {"Time":"2020-09-04T05:08:54.903884762Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/test/e2e/kb","Test":"TestTelemetry/Kibana_should_be_able_to_connect_to_Elasticsearch","Output":"Retries (10m0s timeout): .\n"}
07:08:55  {"Time":"2020-09-04T05:08:54.903919376Z","Action":"run","Package":"github.com/elastic/cloud-on-k8s/test/e2e/kb","Test":"TestTelemetry/Kibana_should_expose_eck_info_in_telemetry_data"}
07:08:55  {"Time":"2020-09-04T05:08:54.903925122Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/test/e2e/kb","Test":"TestTelemetry/Kibana_should_expose_eck_info_in_telemetry_data","Output":"=== RUN   TestTelemetry/Kibana_should_expose_eck_info_in_telemetry_data\n"}
07:08:56  {"Time":"2020-09-04T05:08:56.427281488Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/test/e2e/kb","Test":"TestTelemetry/Kibana_should_expose_eck_info_in_telemetry_data","Output":"Retries (10m0s timeout): .panic: runtime error: index out of range [0] with length 0\n"}
07:08:56  {"Time":"2020-09-04T05:08:56.427353421Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/test/e2e/kb","Test":"TestTelemetry/Kibana_should_expose_eck_info_in_telemetry_data","Output":"\n"}
07:08:56  {"Time":"2020-09-04T05:08:56.42736221Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/test/e2e/kb","Test":"TestTelemetry/Kibana_should_expose_eck_info_in_telemetry_data","Output":"goroutine 7295 [running]:\n"}
07:08:56  {"Time":"2020-09-04T05:08:56.427370898Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/test/e2e/kb","Test":"TestTelemetry/Kibana_should_expose_eck_info_in_telemetry_data","Output":"github.com/elastic/cloud-on-k8s/test/e2e/kb.TestTelemetry.func1.1(0x1babaa0, 0xc000010018)\n"}
07:08:56  {"Time":"2020-09-04T05:08:56.427376749Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/test/e2e/kb","Test":"TestTelemetry/Kibana_should_expose_eck_info_in_telemetry_data","Output":"\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/kb/telemetry_test.go:60 +0x98a\n"}
07:08:56  {"Time":"2020-09-04T05:08:56.427381928Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/test/e2e/kb","Test":"TestTelemetry/Kibana_should_expose_eck_info_in_telemetry_data","Output":"github.com/elastic/cloud-on-k8s/test/e2e/test.UntilSuccess.func1.1(0xc001439800, 0x1a68900)\n"}
07:08:56  {"Time":"2020-09-04T05:08:56.427387263Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/test/e2e/kb","Test":"TestTelemetry/Kibana_should_expose_eck_info_in_telemetry_data","Output":"\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/utils.go:81 +0x8f\n"}
07:08:56  {"Time":"2020-09-04T05:08:56.427392326Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/test/e2e/kb","Test":"TestTelemetry/Kibana_should_expose_eck_info_in_telemetry_data","Output":"github.com/elastic/cloud-on-k8s/pkg/utils/retry.UntilSuccess.func2(0xc0020a6120, 0xc000dc92e0)\n"}
07:08:56  {"Time":"2020-09-04T05:08:56.427403646Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/test/e2e/kb","Test":"TestTelemetry/Kibana_should_expose_eck_info_in_telemetry_data","Output":"\t/go/src/github.com/elastic/cloud-on-k8s/pkg/utils/retry/retry.go:41 +0x27\n"}
07:08:56  {"Time":"2020-09-04T05:08:56.427409546Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/test/e2e/kb","Test":"TestTelemetry/Kibana_should_expose_eck_info_in_telemetry_data","Output":"created by github.com/elastic/cloud-on-k8s/pkg/utils/retry.UntilSuccess\n"}
07:08:56  {"Time":"2020-09-04T05:08:56.427414883Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/test/e2e/kb","Test":"TestTelemetry/Kibana_should_expose_eck_info_in_telemetry_data","Output":"\t/go/src/github.com/elastic/cloud-on-k8s/pkg/utils/retry/retry.go:40 +0xe5\n"}
07:08:56  {"Time":"2020-09-04T05:08:56.431979073Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/test/e2e/kb","Output":"FAIL\tgithub.com/elastic/cloud-on-k8s/test/e2e/kb\t1318.482s\n"}
07:08:56  {"Time":"2020-09-04T05:08:56.43211018Z","Action":"fail","Package":"github.com/elastic/cloud-on-k8s/test/e2e/kb","Elapsed":1318.482}
07:08:56  {"log.level":"info","@timestamp":"2020-09-04T05:08:56.683Z","log.logger":"e2e-fx0oo","message":"Pod is in failed state","service.version":"0.0.0-SNAPSHOT+00000000","service.type":"eck","ecs.version":"1.4.0","name":"eck-e2e-fx0oo-lnt7w"}
07:08:57  {"log.level":"info","@timestamp":"2020-09-04T05:08:57.485Z","log.logger":"e2e-fx0oo","message":"Log stream ended","service.version":"0.0.0-SNAPSHOT+00000000","service.type":"eck","ecs.version":"1.4.0"}
07:08:57  {"log.level":"info","@timestamp":"2020-09-04T05:08:57.485Z","log.logger":"e2e-fx0oo","message":"Log stream stopped","service.version":"0.0.0-SNAPSHOT+00000000","service.type":"eck","ecs.version":"1.4.0"}
```

### PR DESCRIPTION
**Pull request by thbkrkr**
Under unknown conditions, the Kibana telemetry API call may return an empty response and
causes a panic. This commit fixes this by retrying the call until the slice is not empty.

Resolves #3709.

### CODE REVIEW

File: test/e2e/kb/telemetry_test.go [hunk 1]
```diff
@@ -57,6 +59,9 @@ func TestTelemetry(t *testing.T) {
 					if err != nil {
 						return err
 					}
+					if stats == nil || len(stats) == 0 {
```
Comment (sebgl) [inline test/e2e/kb/telemetry_test.go:62]: I think the second condition is technically enough (also works on pointers)

**Review by sebgl — APPROVED** [review]
Lgtm

**Comment by anyasabo** [comment]
Might also be worth opening a kibana issue?

**Review by barkbay — APPROVED** [review]

File: test/e2e/kb/telemetry_test.go (hunk 1 above)
Comment (thbkrkr) [reply test/e2e/kb/telemetry_test.go]: Yes!

**Comment by thbkrkr** [comment]
> Might also be worth opening a kibana issue if non-error responses are always supposed to have a body?

I'm going to reach Kibana folks and see if it's worth opening an issue as I'm not comfortable opening it without being able to repro it.

**Comment by thbkrkr** [comment]
There is a change in the Kibana's startup behaviour that can lead to have `/api/status` to return green while Kibana is not completely ready. The Beats team encountered this issue (https://github.com/elastic/beats/pull/20772#discussion_r482013573). It will be fixed soon (https://github.com/elastic/kibana/pull/76054).

**Pull request merged**